script/cpplint.js

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env node
+
+const { GitProcess } = require('dugite')
+const childProcess = require('child_process')
+const fs = require('fs')
+const klaw = require('klaw')
+const minimist = require('minimist')
+const path = require('path')
+
+const SOURCE_ROOT = path.normalize(path.dirname(__dirname))
+const LINTER_PATH = path.join(SOURCE_ROOT, 'vendor', 'depot_tools', 'cpplint.py')
+const ROOTS = ['atom', 'brightray'].map(token => path.join(SOURCE_ROOT, token))
+
+function isBootstrapped () {
+  return fs.existsSync(LINTER_PATH)
+}
+
+function callCpplint (filenames, args) {
+  if (args.verbose) console.log([LINTER_PATH, ...filenames].join(' '))
+  try {
+    console.log(String(childProcess.execFileSync(LINTER_PATH, filenames, {cwd: SOURCE_ROOT})))
+  } catch (e) {
+    process.exit(1)
+  }
+}
+
+function parseCommandLine () {
+  let help
+  const opts = minimist(process.argv.slice(2), {
+    boolean: ['help', 'onlyChanged', 'verbose'],
+    default: { help: false, onlyChanged: false, verbose: false },
+    alias: { c: 'onlyChanged', 'only-changed': 'onlyChanged', h: 'help', v: 'verbose' },
+    unknown: arg => { help = true }
+  })
+  if (help || opts.help) {
+    console.log('Usage: cpplint.js [-h|--help] [-c|--only-changed] [-v|--verbose]')
+    process.exit(0)
+  }
+  return opts
+}
+
+async function findChangedFiles (top) {
+  const result = await GitProcess.exec(['diff', '--name-only'], top)
+  if (result.exitCode === 0) return new Set(result.stdout.split(/\r\n|\r|\n/g).map(x => path.join(top, x)))
+  console.log('Failed to find changed files', GitProcess.parseError(result.stderr))
+  process.exit(1)
+}
+
+async function findFiles (top, test) {
+  return new Promise((resolve, reject) => {
+    const matches = []
+    klaw(top)
+      .on('end', () => resolve(matches))
+      .on('data', item => {
+        if (test(item.path)) matches.push(item.path)
+      })
+  })
+}
+
+function isCCFile (filename) {
```

Comment:
same here, if it was used more than once i'd say it'd be better to abstract out but this is probably simpler as a one-liner inside the calling function